src/shapes/polygon.cpp

```diff
@@ -0,0 +1,45 @@
+#include "math/pi.hpp"
+#include "shapes/polygon.hpp"
+#include "shapes/star.hpp"
+#include "shapes/straight_vertex.hpp"
+#include <cmath>
+
+using namespace rive;
+
+void Polygon::cornerRadiusChanged() { markPathDirty(); }
+
+void Polygon::pointsChanged() { markPathDirty(); }
+
+void Polygon::clearVertices()
+{
+	for (auto vertex: m_Vertices)
+	{
+		delete vertex;
+	}
+	m_Vertices.clear();
+}
+
+void Polygon::update(ComponentDirt value)
+{
+	if (hasDirt(value, ComponentDirt::Path))
+	{
+		clearVertices();
```

Comment:
Is `update()` called every frame? If so, are we concerned about the performance of rebuilding the vertices index everytime?

I'm thinking since the number of points can't change on either shape (or can they...) that we could probably initialise the vector once to the correct length and just update the values in a loop on each call to update. Would that be better?

Reply:
It's called every frame but it'll only find dirt if the path has been marked dirty, so it'll only re-generate when the parameters have changed. So, yes, if you're animating a property of the star it'll be re-generating the vertices every frame.

I think we will let you animate the number of points, but that'll be a case you can detect. I think it's a good idea to not clear them unless the point() count is different from m_Vertices.size().

Reply:
So you could have a "resize" check that adds/deletes as necessary and only runs if the desired length is different from stored. Then iterate and set x/y/radius but no need to new/add just do m_Vertices[i].

Reply:
Now check for the size and added a `resizeVertices` function to handle this efficiently.